### ISSUE TITLE
Modal: Does not close after pressing `Esc`

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#2404]
Steps to reproduce:

Open Mindtap on Stage/Prod (NOT QA/Int), open any MAST/MTAU/LRNO whatever-we-call-them course. I am using MTPN263ZKMTL but whatever.
Open any MT modal window, for example click on three dots on the right of the activity and select Move Activity in the dropdown. 
Try closing the modal with Esc button.

Expected behavior: modal closes.
Actual behavior: we see an error in the console (fig 1) and the modal does not close.

What's going on here, especially since the issue is not present on QA/Int? On Stage and Prod we have Optanon scripts enabled, which add their own modal window to consent or reject cookies. The modal window is hidden via styles, but still present in DOM (fig 2).
So then when react-magma tries to close the modal it triggers a specific [branch of code](https://github.com/cengage/react-magma/blob/dev/packages/react-magma-dom/src/components/Modal/Modal.tsx#L343): there was one modal when the current one was opened, but now we have two modals in DOM so we get to the error line document.getElementById(id).contains .
If we debug the script (fig 3) we see that the issue appears because id (variable x  in minified code) is undefined (and contentId which is X is "undefined_content").

Now further is my speculation but what I think happens is incorrect usage of hooks.  The [useEffect that adds event listener](https://github.com/cengage/react-magma/blob/dev/packages/react-magma-dom/src/components/Modal/Modal.tsx#L314) only has isModalOpen in dependencies (and worryingly not isEscKeyDownDisabled which it should) so the effect only fires once, because isModalOpen is true in initial render, and adds event listener handleEscapeKeyDown with the initial closure that captured initial id variable. Which I think is a problem because ID generation is also [done in useEffect](https://github.com/cengage/react-magma/blob/dev/packages/react-magma-dom/src/utils/index.ts#L9), after the component's initial render. And because we don't reattach event listener when its dependencies change, we're stuck with initial closure with id = undefined.

To summarize: when we render Modal with isModalOpen: true since the start, the handleEscapeKeyDown should capture actual id and not undefined  so that it doesn't break when there are more modals than react-magma remembers. (Also I'm pretty sure TS should warn you getElementById returns HTMLElement or null so there's something finicky going on anyway). (edited)

**Screenshots:**

<img width="773" height="1195" alt="Image" src="https://github.com/user-attachments/assets/66174dbe-d8da-4f37-9749-24aca28a36e6" />
<img width="770" height="415" alt="Image" src="https://github.com/user-attachments/assets/bb8c3176-399e-4fb9-be51-3056b1bf1b71" />
<img width="712" height="436" alt="Image" src="https://github.com/user-attachments/assets/d2270fe1-e8be-43a6-951d-ac4dc2e2b46a" />

[Slack](https://cengage.slack.com/archives/C0A9TSB75P0/p1775051243754609)

**Comment by artemtrusov-cengage** [comment]
**Tested on:**
- V5: on [Storybook V5](https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/modal--esc-with-foreign-aria-modal)
- V4:  on [Storybook V4](https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/modal--esc-with-foreign-aria-modal)

Everything works as expected. No bugs found.
